tidy3d/plugins/design/method.py

```diff
@@ -1,17 +1,23 @@
 """Defines the methods used for parameter sweep."""
 
 from abc import ABC, abstractmethod
-from typing import Any, Callable, Dict, Tuple, Union
+from typing import Any, Callable, Dict, Literal, Optional, Tuple, Union
 
 import numpy as np
 import pydantic.v1 as pd
+import pygad
 import scipy.stats.qmc as qmc
+from bayes_opt import BayesianOptimization, UtilityFunction
+from bayes_opt.event import Events
+from bayes_opt.logger import JSONLogger
+from pyswarms.single.global_best import GlobalBestPSO
 
-from ... import web
+import tidy3d.plugins.design as tdd
```

Comment:
not sure about this, doesn't it lead to some circular import? since `tdd` imports from this file? surprised this works.